consumer/sm_context.go

```diff
@@ -24,16 +25,67 @@ import (
 	"github.com/omec-project/openapi/models"
 )
 
+func getServingSmfIndex(smfNum int) (servingSmfIndex int) {
+	servingSmfIndexStr := os.Getenv("SERVING_SMF_INDEX")
+	i, _ := strconv.Atoi(servingSmfIndexStr)
+	servingSmfIndexInt := i + 1
+	servingSmfIndex = servingSmfIndexInt % smfNum
+	os.Setenv("SERVING_SMF_INDEX", strconv.Itoa(servingSmfIndex))
+	return
+}
+
+func setAltSmfProfile(smCtxt *amf_context.SmContext) error {
+	ignoreSmfId := smCtxt.SmfID()
+	var altSmfInst []models.NfProfile
+	//iterate over nf instances to ignore failed NF
+	for _, inst := range smCtxt.SmfProfiles {
+		if inst.NfInstanceId != ignoreSmfId {
+			altSmfInst = append(altSmfInst, inst)
+		}
+	}
+
+	if len(altSmfInst) > 0 {
+		smCtxt.SmfProfiles = altSmfInst
+		nfProfile := altSmfInst[0]
+		smfUri := util.SearchNFServiceUri(nfProfile, models.ServiceName_NSMF_PDUSESSION, models.NfServiceStatus_REGISTERED)
+		smCtxt.SetSmfID(nfProfile.NfInstanceId)
+		smCtxt.SetSmfUri(smfUri)
+		return nil
+	}
+	return fmt.Errorf("no alternate profiles available")
+}
+
+func refreshSmfProfiles(ue *amf_context.AmfUe, smCtxt *amf_context.SmContext, ignoreSmfId string) *[]models.NfProfile {
+
+	nrfUri := ue.ServingAMF().NrfUri
+	param := Nnrf_NFDiscovery.SearchNFInstancesParamOpts{
+		ServiceNames: optional.NewInterface([]models.ServiceName{models.ServiceName_NSMF_PDUSESSION}),
+		Dnn:          optional.NewString(smCtxt.Dnn()),
+		Snssais:      optional.NewInterface(util.MarshToJsonString([]models.Snssai{smCtxt.Snssai()})),
+	}
+
+	result, err := SendSearchNFInstances(nrfUri, models.NfType_SMF, models.NfType_AMF, &param)
+	if err != nil {
+		return nil
```

Comment:
Log the error.

Reply:
This code not used as of now and shall be later employed for NRF lib. So, bound to change.